### PR TITLE
feat(documents): added documentSize

### PR DESCRIPTION
**Pull request by AnuragNagpure**
## Description

POST Endpoint to be adjusted: /api/registration/application/{applicationId}/documentType/{documentTypeId}/documents
API should store Length in Kilobyte.
New Databasefield in table documents -> document_size type Long

GET Endpoints to be adjusted:
/api/registration/application/{applicationId}/documentType/{documentTypeId}/documents

Execute SQL script to update the documents.document_size field for existing entries.

## Why

New documentSize field added in docuents table also script for already existing document to get its size in database added.
Post/Get endpoints associated with documents table enhanced.

## Issue

#880 

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Repositories/IDocumentRepository.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public interface IDocumentRepository
     /// <param name="documentTypeId">the document type id</param>
     /// <param name="setupOptionalFields">Action to setup the additional fields</param>
     /// <returns>Returns the created document</returns>
-    Document CreateDocument(string documentName, byte[] documentContent, byte[] hash, MediaTypeId mediaTypeId, DocumentTypeId documentTypeId, Action<Document>? setupOptionalFields);
+    Document CreateDocument(string documentName, byte[] documentContent, byte[] hash, MediaTypeId mediaTypeId, DocumentTypeId documentTypeId, long documentLength, Action<Document>? setupOptionalFields);
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/IDocumentRepository.cs:42]: Please add to the param collection in the summary

File: src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231115.cs [hunk 2]
```diff
@@ -45,6 +45,8 @@ public class AuditDocument20231115 : IAuditEntityV1
 
     public DocumentStatusId? DocumentStatusId { get; set; }
 
+    public long? DocumentSize { get; set; }
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231115.cs:48]: Please never update Audit entities, the way to go would be to create a new entity with the added property and link it appropriately in the origin engine.

File: src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231108.cs [hunk 3]
```diff
@@ -53,6 +53,8 @@ public AuditDocument20231108()
 
     public DocumentStatusId DocumentStatusId { get; set; }
 
+    public long? DocumentSize { get; set; }
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231108.cs:56]: Please never update Audit entities, the way to go would be to create a new entity with the added property and link it appropriately in the origin engine.

File: src/portalbackend/PortalBackend.PortalEntities/Entities/Document.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ private Document()
         CompanyCertificates = new HashSet<CompanyCertificate>();
     }
 
-    public Document(Guid id, byte[] documentContent, byte[] documentHash, string documentName, MediaTypeId mediaTypeId, DateTimeOffset dateCreated, DocumentStatusId documentStatusId, DocumentTypeId documentTypeId)
+    public Document(Guid id, byte[] documentContent, byte[] documentHash, string documentName, MediaTypeId mediaTypeId, DateTimeOffset dateCreated, DocumentStatusId documentStatusId, DocumentTypeId documentTypeId, long documentSize)
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.PortalEntities/Entities/Document.cs:44]: Please adjust the AuditEntityV1 to a newly created auditEntity

File: src/portalbackend/PortalBackend.Migrations/Migrations/20241119105520_document_size_parameter.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+#pragma warning disable CA1814 // Prefer jagged arrays over multidimensional
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.Migrations.Migrations
+{
+    /// <inheritdoc />
+    public partial class document_size_parameter : Migration
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20241119105520_document_size_parameter.cs:29]: Please recreate the migration after adding the new audit entity

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.Migrations/Migrations/20241119105520_document_size_parameter.cs (hunk 5 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20241119105520_document_size_parameter.cs:29]: new migration added

File: src/portalbackend/PortalBackend.PortalEntities/Entities/Document.cs (hunk 4 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/Document.cs:44]: adjusted

File: src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231108.cs (hunk 3 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231108.cs]: sure will note it

File: src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231115.cs (hunk 2 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.PortalEntities/AuditEntities/AuditDocument20231115.cs]: will do the same

File: src/portalbackend/PortalBackend.DBAccess/Repositories/IDocumentRepository.cs (hunk 1 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/IDocumentRepository.cs:43]: added

File: src/portalbackend/PortalBackend.Migrations/Migrations/20241120091826_document_size_field_update_audit_entities.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
+/********************************************************************************
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20241120091826_document_size_field_update_audit_entities.cs:1]: It looks like you didn't rollback the previous migration or didn't rollback the snapshot file, the migration doesn't include the add of the column.

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.Migrations/Migrations/20241120091826_document_size_field_update_audit_entities.cs (hunk 6 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20241120091826_document_size_field_update_audit_entities.cs]: changes done

**Review by Phil91 — APPROVED** [review]

**Pull request merged**